stdlib/public/Python/NumpyConversion.swift

```diff
@@ -0,0 +1,180 @@
+//===-- NumpyConversion.swift ---------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the `NumpyArrayConvertible` protocol for bridging
+// `numpy.ndarray`.
+//
+//===----------------------------------------------------------------------===//
+
+/// A type that can be initialized from a `numpy.ndarray` instance represented
+// as a `PyValue`.
+public protocol NumpyArrayConvertible {
+  init?(numpyArray: PyValue)
+}
+
```

Comment:
Once you find a proper name for the protocol that defines `init?(numpyArray:)`, define another protocol `NumpyArrayConvertible` that defines `var numpyArray: PyValue { get }`.

Reply:
Yep, I thought about this as well.
I started experimenting with `NumpyArrayConvertible` earlier. Initializing `numpy.ndarray` from a pointer is a bit more complicated and involves [`numpy.ctypeslib.as_array`](https://docs.scipy.org/doc/numpy-1.14.0/reference/routines.ctypeslib.html#numpy.ctypeslib.as_array).